### DISCUSSION TITLE
Shrine 2.X with Ruby 3.1  (Wrong Number of Arguments Error)

### DISCUSSION THREAD

**Discussion by aipack** [shrinerb/shrine#702] [General]
Due to Heroku's requirement for new builds or updated apps to use Ruby 3.1, I am forced to upgrade my Rails app (which is currently using Shrine gem 2.19) from Ruby 2.7 to Ruby 3.1. 

This worked perfectly in Ruby 2.7, but after upgrading to Ruby 3.1, 
`<%= image_tag(recent_project.cover_image_url(:large)>`
throws error message:
`ActionView::Template::Error (wrong number of arguments (given 3, expected 2))`

From what I've read, maybe this issue has to do with how Ruby 3.x handles keyword arguments. I don't quite understand, and what's the best way to resolve this situation? Thank you

**Comment by janko** [comment]
Yeah, Ruby 3.0 introduced backwards incompatible keyword argument changes. Shrine 3.x has all
the keyword argument compatibility fixes. What's preventing you from upgrading to Shrine 3.x?

**Reply by aipack** [reply]
---- What's preventing you from upgrading to Shrine 3.x?

Well, it's probably the learning curve of adjusting my code to work with Shrine 3.x.
I wrote the app about five years ago, so I've kind of forgotten how to set up Shrine. I just upgraded to Shrine 3.x, and now there's a bunch of errors waiting for me. :). 

![Screenshot 2024-09-05 at 20 38 01](https://github.com/user-attachments/assets/0e8e6941-c633-43c4-996a-1a3d6fbdc720)

If the recommendation is to upgrade to Shrine 3.x, I guess I'll just have to deal with these errors. Thank you!

**Comment by jrochkind** [comment]
While I think the Shrine team isn't maintaining Shrine 2.x, perhaps if someone else wanted to submit a PR for ruby 3.1 compat, it could be reviewed? i would volunteer to help peer review! 

But def your best best is moving to shrine 3.
